src/components/datePicker/index.tsx

```diff
@@ -73,6 +73,15 @@ const DatePicker: FC<Props> = ({
     }
   }
 
+  const updateFormValue = (newValue) => {
+    callback({
+      target: {
+        name: formFiledName,
+        value: newValue,
+      },
+    })
+  }
```

Comment:
@gzimh inspiration related to your task with the controlled/uncontrolled delayed inputs. maybe this approach works for clearing in your case too and you don't have to fiddle with refs?

Reply:
While on a formik context the onChange its provided by it, the formik value gets updated to empty when the clear button is pressed but the input value is still left there because there is no way to update the value of an uncontrolled input (except via refs). 

I made this implementation with refs which does combine the incoming ref if it exists with an internal input one. So, even if the ref is passed from the consumer apps, they will still work fine (no breaking changes):
https://github.com/carforyou/carforyou-components-pkg/pull/370/commits/42f1c65639d0afcba2cd472cbd99c9cdfac2e664